pkg/dockerfile/generator.go

```diff
@@ -391,9 +390,19 @@ RUN --mount=type=cache,target=/var/cache/apt,sharing=locked apt-get update -qq &
 }
 
 func (g *Generator) installCog() (string, error) {
-	// Wheel name needs to be full format otherwise pip refuses to install it
-	cogFilename := "cog-0.0.1.dev-py3-none-any.whl"
-	lines, containerPath, err := g.writeTemp(cogFilename, cogWheelEmbed)
+	files, err := cogEmbed.ReadDir("embed")
+	if err != nil {
+		return "", err
+	}
+	if len(files) != 1 {
+		return "", fmt.Errorf("should only have one cog wheel embedded")
+	}
+	filename := files[0].Name()
+	data, err := cogEmbed.ReadFile("embed/" + filename)
+	if err != nil {
+		return "", err
+	}
+	lines, containerPath, err := g.writeTemp(filename, data)
```

Comment:
what's the rationale for reading and then writing it like this instead of the previous approach? just having the wheel name?

Reply:
You can't embed a wildcard pattern `embed/*.whl` except into an `embed.FS`, which then requires that we extract it like this. And yes, the advantage is that we can use the built wheel output of `python -m build` without modification or renaming.